src/software/ai/navigator/path_planner/path_planner_test.cpp

```diff
@@ -0,0 +1,239 @@
+
+using namespace Test;
+
+using PathPlannerConstructor = std::function<std::unique_ptr<PathPlanner>()>;
+
+struct PlannerTestCase
+{
+    std::string name = "Unnamed test case";
+    Point start, dest;
+    Rectangle navigable_area;
+    std::vector<Obstacle> obstacles;
+    bool should_return_path;
+};
+
+std::vector<PlannerTestCase>
+    test_cases =
+        {{.name               = "Empty field straight line",
+          .start              = Point(0, 0),
+          .dest               = Point(1, 0),
+          .navigable_area     = Rectangle({-1, -1}, {1, 1}),
+          .obstacles          = {},
+          .should_return_path = true},
+
+         {.name           = "Single stationary robot in path",
+          .start          = Point(0, 0),
+          .dest           = Point(2, 0),
+          .navigable_area = Rectangle({-1, -1}, {1, 1}),
+          .obstacles = {Obstacle::createRobotObstacle(TestUtil::createRobotAtPos({1, 0}),
+                                                      false)},
+          .should_return_path = true},
+
+         {.name               = "Large rectangle in path",
+          .start              = Point(-3, 0),
+          .dest               = Point(4, 0),
+          .navigable_area     = Rectangle({-5, -5}, {5, 5}),
+          .obstacles          = {Obstacle(Rectangle({1, 4}, {2, -4}))},
+          .should_return_path = true},
+
+         {.name           = "Circle of robots surrounding ego at distance 1",
+          .start          = Point(0, 0),
+          .dest           = Point(4, 0),
+          .navigable_area = Rectangle({-5, -5}, {5, 5}),
+          .obstacles =
+              {
+                  Obstacle::createRobotObstacle(TestUtil::createRobotAtPos({1, 0}),
+                                                false),
+                  Obstacle::createRobotObstacle(
+                      TestUtil::createRobotAtPos(
+                          {std::cos(M_PI / 3), std::sin(M_PI / 3)}),
+                      false),
+                  Obstacle::createRobotObstacle(
+                      TestUtil::createRobotAtPos(
+                          {std::cos(2 * M_PI / 3), std::sin(2 * M_PI / 3)}),
+                      false),
+                  Obstacle::createRobotObstacle(
+                      TestUtil::createRobotAtPos(
+                          {std::cos(3 * M_PI / 3), std::sin(3 * M_PI / 3)}),
+                      false),
+                  Obstacle::createRobotObstacle(
+                      TestUtil::createRobotAtPos(
+                          {std::cos(4 * M_PI / 3), std::sin(4 * M_PI / 3)}),
+                      false),
+                  Obstacle::createRobotObstacle(
+                      TestUtil::createRobotAtPos(
+                          {std::cos(5 * M_PI / 3), std::sin(5 * M_PI / 3)}),
+                      false),
+              },
+          .should_return_path = true},
+
+         {.name           = "Circle of robots surrounding ego at distance 0.2",
```

Comment:
"ego" - I see someone has been working with autonomous cars in the valley...... :joy:

Reply:
But actually, we should probably change "ego" to "a friendly robot", because most people won't get that lol

Reply:
removed